lib/services/local/notifications.go

```diff
@@ -236,6 +236,10 @@ func (s *NotificationsService) CreateGlobalNotification(ctx context.Context, glo
 		return nil, trace.Wrap(err)
 	}
 
+	// Set the same expiry on the outer GlobalNotification wrapper's metadata. This is necessary for the sqlite cleanup routine
```

Comment:
I would do this up on line 232 where you set the name.

Reply:
this has to be done after the `CheckAndSetExpiry()`, so I'll move those lines where we set the name down instead

Reply:
I was thinking more like this, so the expiry is never left unset.
```go
globalNotification.Metadata = &headerv1.Metadata{
    Name:    uuid.String(),
    Expires: ...
}
```